src/core/Akka/Actor/Address.cs

```diff
@@ -256,7 +258,8 @@ public Address WithPort(int? port = null)
         /// <returns><c>true</c> if both addresses are not equal; otherwise <c>false</c></returns>
         public static bool operator !=(Address left, Address right)
         {
-            return !Equals(left, right);
+            return !ReferenceEquals(left, right) &&
+                   (left?.Equals(right) ?? true);
```

Comment:
This looks incorrect. if `left` equals `right` we should return false. (I think the null coalesce is making it hard to infer here)

Reply:
Yeah based on the test suite I probably did that wrong

Reply:
Ah, I thought I was !-ing the entire statement, not just the first clause.